typescript/ica/src/deploy.ts

```diff
@@ -50,3 +56,40 @@ export class InterchainAccountDeployer<
     };
   }
 }
+
+export type InterchainQueryConfig = RouterConfig;
+
+export class InterchainQueryDeployer<
+  Chain extends ChainName,
+> extends HyperlaneRouterDeployer<
+  Chain,
+  InterchainQueryConfig,
+  InterchainQueryContracts,
+  InterchainQueryFactories
+> {
+  constructor(
+    multiProvider: MultiProvider<Chain>,
+    configMap: ChainMap<Chain, InterchainQueryConfig>,
+    protected core: HyperlaneCore<Chain>,
+    protected salt = 'asdasdsd',
```

Comment:
`salt` seems like too generic for this use case, probably would name it `create2Salt`. I would even be in favor of moving this to a parent class and then actually have  a "per-app base salt" and then a "per-contract" salt as well that gets combined